client/client.go

```diff
@@ -49,6 +51,11 @@ func NewClient(url, user, password string, verifyTLS bool, caCertByte []byte) *C
 		gorequest: gorequest.New().TLSClientConfig(tlsConfig),
 	}
 
+	proxy := os.Getenv("https_proxy")
```

Comment:
does the env not supposed to be with capital letters?
`HTTPS_PROXY`

Reply:
Terraform Enterprise exposes it only via lowercase letters if its configured in the platform. But yes, users are able to inject arbitrary env variables inside a TF run.

Reply:
I will rework it and use "net/url" and "golang.org/x/net/http/httpproxy" so we implement what users normally use :-)

So we cover all (edge-)cases
- http_proxy / HTTP_PROXY
- https_proxy /HTTPS_PROXY
- no_proxy / NO_PROXY

Reply:
Updated PR